near/omni-bridge/src/lib.rs

```diff
@@ -517,11 +547,122 @@ impl Contract {
             )
             .into()
         } else {
-            self.process_fin_transfer_to_other_cahin(predecessor_account_id, transfer_message);
+            self.process_fin_transfer_to_other_chain(predecessor_account_id, transfer_message);
             PromiseOrValue::Value(destination_nonce)
         }
     }
 
+    fn fast_fin_transfer(
+        &mut self,
+        sender_id: AccountId,
+        token_id: AccountId,
+        amount: U128,
+        fast_fin_transfer_msg: FastFinTransferMsg,
+    ) -> PromiseOrValue<U128> {
+        let fast_transfer = FastTransfer {
+            token_id: token_id.clone(),
+            recipient: fast_fin_transfer_msg.recipient.clone(),
+            amount: U128(amount.0 + fast_fin_transfer_msg.fee.fee.0),
+            fee: fast_fin_transfer_msg.fee,
```

Comment:
Who will ultimately receive the native fee? The Fast Transfer Relayer or the General Relayer? Are we at risk of spending the same funds twice?

Reply:
Perhaps we can allow `sign_transfer` only for the fast relayer in this case

Reply:
Actually, in case recipient is on another chain we need a solution for both normal and fast transfers. So perhaps it's not quite in scope of this PR. As of now, the relayer that finalizes the transfer on the destination chain will receive the fees